cshared.go

```diff
@@ -116,10 +133,40 @@ func FLBPluginInit(ptr unsafe.Pointer) int {
 	return input.FLB_OK
 }
 
-// FLBPluginInputCallback this method gets invoked by the fluent-bit runtime, once the plugin has been
-// initialised, the plugin implementation is responsible for handling the incoming data and the context
-// that gets past, for long-living collectors the plugin itself should keep a running thread and fluent-bit
-// will not execute further callbacks.
+// flbPluginReset is meant to reset the plugin between tests.
+func flbPluginReset() {
+	theInputLock.Lock()
+	defer theInputLock.Unlock()
+
+	once = sync.Once{}
```

Comment:
Oh, `once sync.Once{}` can be re-initialized!?? I didn't know about that.
This is just a comment no need to take care of.

Reply:
I mean, you probably shouldn't, but we did 😆 . That same function should be usable for hot-reloading. I'll test #51 again now that it's been rebased.